build_cmake/Framework/CMakeLists.txt

```diff
@@ -0,0 +1,148 @@
+cmake_minimum_required(VERSION 2.8)
+project(Framework)
+
+set(CMAKE_MODULE_PATH
+	${CMAKE_CURRENT_SOURCE_DIR}/../../../Dependencies/cmake-modules
+	${CMAKE_MODULE_PATH}
+)
+
+if(NOT CMAKE_BUILD_TYPE)
+	set(CMAKE_BUILD_TYPE Release CACHE STRING
+		"Choose the type of build, options are: None Debug Release"
+		FORCE)
+endif()
+
+if(CMAKE_BUILD_TYPE STREQUAL "Release")
```

Comment:
@Thunder07 I was thinking of replacing this with something that adds the proper preprocessor flags to `CMAKE_CXX_FLAGS_DEBUG` and `CMAKE_CXX_FLAGS_RELEASE` since the current thing won't work with multi-configuration project types (on Xcode and MSVC). Do you see any problems with that?